membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -215,5 +221,40 @@ object AttributesFromZuora {
 
     regular.map(_.copy(DigitalSubscriptionExpiryDate = maybeLatestDate)).orElse(gift)
   }
+
+  val metrics = Metrics("AttributesFromZuora") //referenced in CloudFormation
+
+  def alertIfAttributesDoNotMatch(fromZuora: Option[Attributes], fromSupporterProductData: Option[Attributes]): Unit =
+    if (attributesDoNotMatch(fromZuora, fromSupporterProductData)) {
+      logger.warn(
+        "There was a mismatch between the attributes returned by Zuora and the supporter-product-data data store\n" +
+        s"Zuora returned: $fromZuora\n" +
+        s"supporter-product-data returned: $fromSupporterProductData"
+      )
+      metrics.put("AttributesMismatch", 1) //referenced in CloudFormation
+    } else logger.info("attributes returned from Zuora match those returned from supporter-product-data")
+
+  def attributesDoNotMatch(fromZuora: Option[Attributes], fromSupporterProductData: Option[Attributes]) =
```

Comment:
why not change this to attributesMatch, then flip the order of the if statment, just to get rid of the negative

Reply:
I preferred `if (attributesDoNotMatch(fromZuora, fromSupporterProductData))` in the main logic flow but it's marginal

Reply:
I mean don't use `!` at all - you can just flip the if and else parts of the statement.

Reply:
Oh I see, I wasn't actually going to have an else clause originally because it could be very noisy in the logs, but then I wanted to be sure that the code was executing successfully so added it at the last moment

Reply:
fair enough, in that case leave it as it is.  This code won't be here long anyway.